solr/modules/analysis-extras/build.gradle

```diff
@@ -15,25 +15,33 @@
  * limitations under the License.
  */
 
-
 apply plugin: 'java-library'
 
 description = 'Additional analysis components'
 
 dependencies {
   api project(':solr:core')
 
+  implementation project(':solr:solrj')
+
   implementation 'commons-io:commons-io'
-  implementation "org.apache.lucene:lucene-analysis-icu"
-  implementation "org.apache.lucene:lucene-analysis-smartcn"
-  implementation "org.apache.lucene:lucene-analysis-morfologik"
-  implementation "org.apache.lucene:lucene-analysis-opennlp"
-  implementation "org.apache.lucene:lucene-analysis-smartcn"
-  implementation "org.apache.lucene:lucene-analysis-stempel"
+  implementation 'com.ibm.icu:icu4j'
+  implementation 'org.apache.lucene:lucene-analysis-icu'
+  implementation 'org.apache.lucene:lucene-analysis-smartcn'
+  implementation 'org.apache.lucene:lucene-analysis-morfologik'
+  implementation 'org.apache.lucene:lucene-analysis-opennlp'
+  implementation 'org.apache.lucene:lucene-analysis-smartcn'
+  implementation 'org.apache.lucene:lucene-analysis-stempel'
+  implementation 'org.apache.lucene:lucene-core'
+  implementation 'org.apache.opennlp:opennlp-tools'
+  implementation 'org.slf4j:slf4j-api'
 
   testImplementation project(':solr:test-framework')
+  testImplementation 'org.apache.lucene:lucene-analysis-common'
+  testImplementation 'junit:junit'
   testImplementation('org.mockito:mockito-core', {
     exclude group: "net.bytebuddy", module: "byte-buddy-agent"
   })
-  testImplementation('org.apache.logging.log4j:log4j-core')
+  testImplementation 'org.apache.logging.log4j:log4j-api'
```

Comment:
This seems wrong to me, since shouldn't we be using slf4j-api everywhere except for the log4j specific log-level handling code in core? Maybe this is a good follow on issue though.

Reply:
Yea its a good question and one to follow up on.

Reply:
This is used in https://github.com/apache/solr/blob/main/solr/modules/analysis-extras/src/test/org/apache/solr/schema/TestICUCollationFieldUDVAS.java to check specific log messages.

Reply:
Yea this looks like something that could be improved separately.

Reply:
Created SOLR-15991